ServerCore/Pages/Puzzles/BulkFiles.cshtml.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Puzzles
+{
+    /// <summary>
+    /// Page for managing all files in bulk
+    /// </summary>
+    [Authorize(Policy = "IsEventAdminOrEventAuthor")]
+    public class BulkFilesModel : EventSpecificPageModel
+    {
+        public BulkFilesModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
+        {
+        }
+
+        [BindProperty]
+        public List<string> FoldersCopied { get; set; } = new List<string>();
+
+        [BindProperty]
+        public List<string> FilesIgnored { get; set; } = new List<string>();
+
+        [BindProperty]
+        public List<IFormFile> NewBulkFiles { get; set; }
+
+        /// <summary>
+        /// Handler for listing files
+        /// </summary>
+        public IActionResult OnGet(int puzzleId)
+        {
+            return Page();
+        }
+
+        /// <summary>
+        /// Handler for uploading files
+        /// </summary>
+        public async Task<IActionResult> OnPostAsync(int puzzleId)
+        {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            if (NewBulkFiles != null)
+            {
+                foreach (IFormFile resourceFile in NewBulkFiles)
+                {
+                    await UploadFileAsync(resourceFile);
+                }
+            }
+
+            return Page();
+        }
+
+        /// <summary>
+        /// Helper for taking an uploaded form file, uploading it, and tracking it in the database
+        /// </summary>
+        private async Task UploadFileAsync(IFormFile uploadedFile)
+        {
+            if (Path.GetExtension(uploadedFile.FileName).Equals(".zip", StringComparison.OrdinalIgnoreCase))
```

Comment:
[question]: So if the file is not a zip, nothing will happen?

Reply:
Correct - I will make it report this via FilesIgnored, for clarity.